manifests/init.pp

```diff
@@ -420,6 +424,9 @@
   validate_array($dhcp_option_domain)
   validate_integer($dhcp_omapi_port)
   validate_string($dhcp_provider, $dhcp_server)
+  if $dhcp_subnets {
```

Comment:
This if statement isn't needed. It can always be an array (and it's the default).